### DISCUSSION TITLE
caplog: Use 'session' scope

### DISCUSSION THREAD

**Discussion by MarBra** [pytest-dev/pytest#11177] [General] [answered]
Hi pytest-experts :) 

i like to use the caplog fixture in another fixture with session scope:

```python
import pytest
import logging

@pytest.fixture(scope='session')
def my_log(caplog):
    caplog.set_level(logging.INFO)

def test_foo(my_log):
    pass
```
This results in the following error:
`ScopeMismatch: You tried to access the 'function' scoped fixture 'caplog' with a 'session' scoped request object, involved factories`

**Comment by RonnyPfannschmidt** [comment]
thats currently not supported

**Comment by Zac-HD** [comment]
You can't use function-scoped fixtures in session-scoped fixtures, because there's no correct time to create or destroy the result!  `caplog` specifically basically *has* to be function-scoped to avoid weird interactions, too.  See e.g. #2704.

Fortunately there's an easy fix here - just define `my_log` as a function-scoped fixture and you'll be fine.

[a comment by an automated account is not shown]

**Comment by nickeldan** [comment]
@Forty-Bot, where does that code get put?

[a comment by an automated account is not shown]

**Comment by caplogebenezere** [comment]
Caplog
